packages/core/utils/src/serializer.js

```diff
@@ -1,5 +1,21 @@
 // @flow
 
+const nameToCtor: Map<string, Class<*>> = new Map();
+const ctorToName: Map<Class<*>, string> = new Map();
+
+export function registerSerializableClass(name: string, ctor: Class<*>) {
```

Comment:
probably could get the name from `constructor.name`?

Reply:
I considered this, however, I always get concerned relying on constructor names in any permanent sense because of minifies.

I guess this would really not impact us here unless we had different builds of the same version of core registering themselves, in which case we'd have other problems.

This would also lock us into needing to use constructed classes for everything, when in the future we could accept another type name property instead. Again maybe not an issue...

Reply:
The other thing is that we need to incorporate other differentiating properties like the version of the constructor.